app/resources/card_resource.rb

```diff
@@ -74,7 +74,8 @@ class CardResource < ApplicationResource # rubocop:disable Metrics/ClassLength
 
     unless @object.num_extra_faces.zero?
       @object.face_indices.each do |index|
-        f = { index:, images: images(@object.latest_printing_id, face_index: index) }
+        f = { index:,
+              images: images(@object.latest_printing_id, @object.printings.last.xlarge_image?, face_index: index) }
```

Comment:
ah, ok.  so there is a lightly tricky thing here.

You removed a read in favor of a different read.

@object.printings does a fetch for all related printings and it is only an accident tht you found this in the last one.  technically they are unordered but sometimes do the right thing.

you want `@object.printing_ids_in_database.first` which will use the array field from the already loaded object.  You can't just use printing_ids because printings is a relationship and printing_ids is a helper for relationships that will give you the ids after a query and loading those associated records.

Reply:
ok, i was too fixated on the "avoid a read" bit and missed the "this doesn't solve your problem" bit.

We will need to pull an extra attribute from the printing into the card view, which i can do quickly as a follow-up.

The actual, helpful, fix you need here is an explicit ordering of the printings or it will sometimes misbehave.

`@object.printings.order(:date_release).reverse_order.first.xlarge_image?` in this case.